signup/signup.js

```diff
@@ -0,0 +1,144 @@
+});
+// 이메일 양식 검증
+function isValidEmail(email) {
+  const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+  return emailRegex.test(email);
+}
+//닉네임
+document.getElementById('username').addEventListener('blur', function(){
+  const usernameInput = document.getElementById('username');
+  const errorMassage = document.getElementById('error-username-message');
+  if(!usernameInput.value) {
+    usernameInput.classList.add('error');
+    errorMassage.style.display = 'block';
+  } else {
+    usernameInput.classList.remove('error');
+    errorMassage.style.display = 'none';
+  }
+  validateForm();
+});
+//비밀번호
+document.getElementById('password').addEventListener('blur', function(){
+  const passwordInput = document.getElementById('password');
+  const errorMassage = document.getElementById('error-password-message');
+  const errorMassageLength = document.getElementById('error-message-length');
+  if (passwordInput.value.trim() ==='') {
+    passwordInput.classList.add('error');
+    errorMassage.style.display = 'block';
+    errorMassageLength.style.display = 'none';
+  } else if (passwordInput.value.length < 8) {
+    passwordInput.classList.add('error');
+    errorMassageLength.style.display = 'block';
+    errorMassage.style.display = 'none';
+  } else {
+    passwordInput.classList.remove('error');
+    errorMassage.style.display = 'none';
+    errorMassageLength.style.display = 'none';
+  }
+  validateForm();
+});
+//비밀번호 8자리 검증
+function isValidPassword(password) {
+  return password.length >=8;
+};
+//비밀번호 재확인
+document.getElementById('checkpassword').addEventListener('blur', function(){
+  checkPasswordMatch();
+  validateForm();
+});
+//비밀번호 확인
+function checkPasswordMatch() {
+  const passwordInput = document.getElementById('password');
+  const checkpassword = document.getElementById('checkpassword');
+  const errorMassageMatch = document.getElementById('error-matchpassword-message');
+  if (passwordInput.value !== checkpassword.value) {
+    checkpassword.classList.add('error');
+    errorMassageMatch.style.display = 'block';
+  } else {
+    checkpassword.classList.remove('error');
+    errorMassageMatch.style.display = 'none';
+  }
+};
+//회원가입 활성화 비활성화
+function validateForm() {
+  const emailInput = document.getElementById('email');
+  const usernameInput = document.getElementById('username');
+  const passwordInput = document.getElementById('password');
+  const checkPasswordInput = document.getElementById('checkpassword');
+  const signupButton = document.getElementById('signup-button');
+
+  const emailValid = isValidEmail(emailInput.value.trim());
+  const usernameValid = usernameInput.value.trim() !== '';
+  const passwordValid = isValidPassword(passwordInput.value.trim());
+  const passwordsMatch = passwordInput.value === checkPasswordInput.value;
+
+  if (emailValid && usernameValid && passwordValid && passwordsMatch) {
+    signupButton.disabled = false;
+  } else {
+    signupButton.disabled = true;
+  }
```

Comment:
💬 여담
이런경우 아래처럼 작성하셔도 됩니다~

```suggestion
  const isFormValid = emailValid && usernameValid && passwordValid && passwordsMatch;
  
  signupButton.disabled = !isFormValid;
```